main.tf

```
@@ -1,88 +1,137 @@
+  is_transient = false
 }
 
-resource "snowflake_warehouse" "warehouse" {
-  provider       = snowflake.sys_admin
-  name           = "TF_DEMO"
-  warehouse_size = "large"
-
+# Create a warehouse
+resource "snowflake_warehouse" "tf_warehouse" {
+  name = "TF_DEMO_WH"
+  warehouse_type = "STANDARD"
+  warehouse_size = "XSMALL"
+  max_cluster_count = 1
+  min_cluster_count = 1
   auto_suspend = 60
+  auto_resume = true
+  enable_query_acceleration = false
+  initially_suspended = true
+}
+
+# Create a new schema in the DB
+resource "snowflake_schema" "tf_db_tf_schema" {
+  name                = "TF_DEMO_SC"
+  database            = snowflake_database.tf_db.name
+  with_managed_access = false
 }
 
+# New provder that will use USERADMIN to create users, roles, and grants
 provider "snowflake" {
-  alias = "security_admin"
-  role  = "SECURITYADMIN"
+    organization_name = "your_org_name"
+    account_name      = "your_account_name"
+    user              = "TERRAFORM_SVC"
+    role              = "USERADMIN"
+    alias             = "useradmin"
+    authenticator     = "SNOWFLAKE_JWT"
+    private_key       = file("~/.ssh/terraform_demo.p8")
 }
 
-resource "snowflake_role" "role" {
-  provider = snowflake.security_admin
-  name     = "TF_DEMO_SVC_ROLE"
+# Create a new role using USERADMIN
+resource "snowflake_account_role" "tf_role" {
+    provider          = snowflake.useradmin
+    name              = "TF_DEMO_ROLE"
+    comment           = "My Terraform role"
 }
 
-resource "snowflake_database_grant" "grant" {
-  provider          = snowflake.security_admin
-  database_name     = snowflake_database.db.name
-  privilege         = "USAGE"
-  roles             = [snowflake_role.role.name]
-  with_grant_option = false
+# Grant the new role to SYSADMIN (best practice)
+resource "snowflake_grant_account_role" "grant_tf_role_to_sysadmin" {
+    provider         = snowflake.useradmin
+    role_name        = snowflake_account_role.tf_role.name
+    parent_role_name = "SYSADMIN"
 }
 
-resource "snowflake_schema" "schema" {
-  provider   = snowflake.sys_admin
-  database   = snowflake_database.db.name
-  name       = "TF_DEMO"
-  is_managed = false
+# Create a key for the new user
+resource "tls_private_key" "svc_key" {
+    algorithm = "RSA"
+    rsa_bits  = 2048
 }
 
-resource "snowflake_schema_grant" "grant" {
-  provider          = snowflake.security_admin
-  database_name     = snowflake_database.db.name
-  schema_name       = snowflake_schema.schema.name
-  privilege         = "USAGE"
-  roles             = [snowflake_role.role.name]
-  with_grant_option = false
+# Create a new user
+resource "snowflake_user" "tf_user" {
+    provider          = snowflake.useradmin
+    name              = "TF_DEMO_USER"
+    default_warehouse = snowflake_warehouse.tf_warehouse.name
+    default_role      = snowflake_account_role.tf_role.name
+    default_namespace = "${snowflake_database.tf_db.name}.${snowflake_schema.tf_db_tf_schema.fully_qualified_name}"
```

I think this should be 
```suggestion
    default_namespace = snowflake_schema.tf_db_tf_schema.fully_qualified_name
```